pyee/util.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+
+from collections import OrderedDict
+
+
+def evolve(cls, underlying, *args, **kwargs):
+    """A helper to create instances of an event emitter ``cls`` that inherit
+    event behavior from an ``underlying`` event emitter instance.
+
+    When called, this instantiates a new instance of ``cls`` and then copies
+    all the handlers on ``underlying`` to that new instance, before returning
+    it. Afterwards, any modifications made to the underlying event emitter
+    and the new "evolved" event emitter are not shared.
+
+    This is mostly helpful if you have a simple underlying event emitter
```

Comment:
What if we make this just `isinstance(underlying, BaseEventEmitter)` and rule out all of these weird edge cases?